src/plugins/vis_type_table/public/table_vis_response_handler.ts

```diff
@@ -49,61 +47,66 @@ export interface TableGroup {
   row: number;
 }
 
-export interface Table {
-  $parent?: TableGroup;
-  columns: Input['columns'];
-  rows: Input['rows'];
+export interface TableContext {
+  table?: Table;
+  tableGroups: TableGroup[];
+  direction?: 'row' | 'column';
 }
 
-export function tableVisResponseHandler(table: Input, dimensions: any): TableContext {
-  const converted: TableContext = {
-    tables: [],
-  };
+export function tableVisResponseHandler(
```

Comment:
This function is pretty meaty. 

1. Can it be simplified and/or broken up in any way to make it more readable?
2. Can we add any unit tests to cover all of these conditionals?

Reply:
Yeah. They are actually from the original angular code [here](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/d7004dc5b0392477fdd54ac66b29d231975a173b/src/plugins/vis_type_table/public/table_vis_response_handler.ts#L58).

We will add some functional tests for this part. I document all the functional tests [here](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2857) and will add these in opensearch-dashboards-functional-test repo. The split table section in the functional test issue will test whether table can be split in rows and columns and the values in the split tables.

Reply:
I'm glad you are adding functional tests, but I think this needs unit tests as well.

Reply:
Yeah functionality is a concern for everyone to review such large PR. 
Functional tests in original OSD are removed and the tests related to table vis are re-created here:
https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/423

Currently, some tests are failed because we need this table vis to hook up to OSD. 
Once table is added in OSD and functionalities are all good, I will start adding unit tests. I also opened a tracking issue.

Reply:
I second @tmarkley's comment about adding a unit test for this function. Looks like a pretty straightforward function with a decent amount of logic.

Reply:
Yes. I have a separate unit test issue and will create all the unit tests.
Here is the issue: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2856.